File: BAEKJOON/java/12919_A와_B_2.java
import java.io.*;

public class Main {
    public static String s, t;

    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        input();
        solution();

        bw.flush();
        bw.close();
    }

    private static void input() throws IOException {
        s = br.readLine();
        t = br.readLine();
    }

    private static void check(String s, String t) {
        if (s.equals(t)) {
            System.out.println(1);
            System.exit(0);
        }

        if (s.length() == t.length()) {
            return;
        }

        if (t.charAt(t.length() - 1) == 'A') {
            StringBuilder tempT = new StringBuilder(t);
            check(s, tempT.deleteCharAt(tempT.length() - 1).toString());
        }

        if (t.charAt(0) == 'B') {
            StringBuilder tempT = new StringBuilder(t);
            tempT.deleteCharAt(0);
            tempT.reverse();

            check(s, tempT.toString());
        }
    }

    private static void solution() throws IOException {
        check(s, t);

        bw.write(String.valueOf(0));
    }
}

File: BAEKJOON/cpp/12919_A와_B_2.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

string s, t;

void input() {
	cin >> s >> t;
}

void check(string s, string t) {
	if (s == t) {
		cout << 1;
		exit(0);
	}

	if (s.size() == t.size()) {
		return;
	}

	if (t[t.size() - 1] == 'A') {
		string temp_t(t);
		check(s, temp_t.erase(temp_t.size() - 1, 1));
	}

	if (t[0] == 'B') {
		string temp_t(t);
		temp_t.erase(0, 1);
		reverse(temp_t.begin(), temp_t.end());

		check(s, temp_t);
	}
}

void solution() {
	check(s, t);

	cout << 0;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	input();
	solution();

	return 0;
}